shared/mathutil/math_helper.go

```diff
@@ -39,3 +41,13 @@ func PowerOf2(n uint64) uint64 {
 	}
 	return 1 << n
 }
+
+// ClosestPowerOf2 returns an integer that is the closest
+// power of 2 that is less than or equal to the argument.
+func ClosestPowerOf2(n uint64) uint64 {
+	if n == 0 {
+		panic("invalid value")
```

Comment:
Should we be using an error instead of panicking here?

Reply:
no